src/Order.Data/Specifications/BaseEmptySpecification.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace Order.Data.Specifications
+{
+    public abstract class BaseEmptySpecification<TEntity, TResult> : ISpecification<TEntity, TResult> where TEntity: class
```

Comment:
If possible I would use some existing implementation of specifications, specification evaluators, generic repositories, etc like https://github.com/ardalis/Specification/.